modules/cudacodec/src/precomp.hpp

```diff
@@ -81,7 +81,7 @@
     #include "video_decoder.hpp"
     #include "video_parser.hpp"
 
-    #include "../src/cap_ffmpeg_api.hpp"
+    #include "cuvid_ffmpeg_api.hpp"
```

Comment:
We need guard: `#if defined(HAVE_FFMPEG)` here and in places where this API is used.

Reply:
I am not that familiar with CMake so I don't know how easy it would be to achieve, but wouldn't it be an idea to forced Build_opencv_cudacodec=FALSE if -DWITH_FFMPEG=OFF in addition to or instead of  placing guards around all the encode decode source files?